<launch>
  <!-- <include file="$(find icreate_navigation)/launch/velocity_smoother.launch.xml"/> to higher level -->

  <arg name="ns" default="" />
  <arg name="name" default=""/>
  <arg name="odom_frame_id"   default="odom"/>
  <arg name="base_frame_id"   default="base_footprint"/>
  <arg name="global_frame_id" default="map"/>
  <arg name="laser_frame_id" default="laser"/>
  <arg name="odom_topic" default="odom" />
  <arg name="laser_topic" default="scan" />
  <arg name="map_topic" default="map" />


  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <rosparam file="$(find icreate_navigation)/params/costmap/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find icreate_navigation)/params/costmap/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find icreate_navigation)/params/costmap/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find icreate_navigation)/params/costmap/global_costmap_params.yaml" command="load" />

    <rosparam file="$(find icreate_navigation)/params/planner/dwa_local_planner_params.yaml" command="load" />
    <rosparam file="$(find icreate_navigation)/params/planner/base_local_planner_params.yaml" command="load" />
    <rosparam file="$(find icreate_navigation)/params/move_base_params.yaml" command="load" />

    <rosparam file="$(find icreate_navigation)/params/planner/global_planner_params.yaml" command="load" />
    <rosparam file="$(find icreate_navigation)/params/planner/navfn_global_planner_params.yaml" command="load" />

    <!-- external params file that could be loaded into the move_base namespace -->
    <!--rosparam file="$(arg custom_param_file)" command="load" /-->

    <!-- reset frame_id parameters using user input data -->
    <param name="global_costmap/global_frame" value="$(arg global_frame_id)"/>
    <param name="global_costmap/robot_base_frame" value="$(arg base_frame_id)"/>
    <param name="local_costmap/global_frame" value="$(arg odom_frame_id)"/>
    <param name="local_costmap/robot_base_frame" value="$(arg base_frame_id)"/>
    <param name="DWAPlannerROS/global_frame_id" value="$(arg odom_frame_id)"/>
    <param name="global_costmap/static_layer/map_topic" value="$(arg map_topic)"/>
    <param name="global_costmap/obstacle_layer/scan/topic" value="$(arg laser_topic)"/>
    <param name="global_costmap/obstacle_layer/scan/sensors_frame" value="$(arg laser_frame_id)"/>
    <param name="local_costmap/obstacle_layer/scan/topic" value="$(arg laser_topic)"/>
    <param name="local_costmap/obstacle_layer/scan/sensors_frame" value="$(arg laser_frame_id)"/>

    <!-- (OUTPUT) from move_base cmd_vel now mapped to velocity_smoother -->
    <remap from="cmd_vel" to="navigation_raw_cmd_vel"/>

    <!-- odometry and laser_scan topics mapping -->
    <!--<remap from="odom" to="$(arg odom_topic)"/>-->
    <!--<remap from="scan" to="$(arg laser_topic)"/>-->
  </node>
</launch>